src/merkle_tree/mod.rs

```diff
@@ -133,12 +135,121 @@ impl Pushable for &MerkleTreeTwinProof {
     }
 }
 
+impl MerkleTreeTwinProof {
+    /// Convert a stwo Merkle proof into twin proofs for each pairs of queries.
+    pub fn from_stwo_proof(
```

Comment:
I feel that this function has a lot of complexity and the code is pretty long. Any way to split in smaller logical parts or you think it's the optimal atomicity ?